tests/sharding/test_vmc_handler.py

```diff
@@ -0,0 +1,167 @@
+            period_start_prevhash,
+            parent_collation_hash,
+            tx_list_root,
+            collation_coinbase,
+            post_state_root,
+            receipt_root,
+            number,
+        ])
+    )
+    sig = vmc_utils.sign(sighash, privkey)
+    return rlp.encode([
+        shard_id,
+        expected_period_number,
+        period_start_prevhash,
+        parent_collation_hash,
+        tx_list_root,
+        collation_coinbase,
+        post_state_root,
+        receipt_root,
+        number,
+        sig,
+    ])
+
+def test_vmc_handler(chain_handler):
+    shard_id = 0
+    validator_index = 0
+    primary_addr = keys[validator_index].public_key.to_checksum_address()
+    zero_addr = eth_utils.address.to_checksum_address(b'\x00' * 20)
+
+    vmc_handler = VMCHandler(chain_handler, primary_addr=primary_addr)
+    print(
+        "!@# viper_rlp_decoder_addr:",
+        eth_utils.to_checksum_address(vmc_utils.viper_rlp_decoder_addr),
+    )
+    print(
+        "!@# sighasher_addr:",
+        eth_utils.to_checksum_address(vmc_utils.sighasher_addr)
+    )
+
+    if not vmc_handler.is_vmc_deployed():
+        print('not handler.is_vmc_deployed()')
+        # import privkey
+        for key in keys:
+            vmc_handler.import_key_to_chain_handler(key)
+
+        vmc_handler.deploy_initiating_contracts(keys[validator_index])
+        vmc_handler.chain_handler.mine(1)
+        vmc_handler.first_setup_and_deposit(keys[validator_index])
+
+    assert vmc_handler.is_vmc_deployed()
+
+    vmc_handler.chain_handler.mine(SHUFFLING_CYCLE_LENGTH)
+    # handler.deploy_valcode_and_deposit(validator_index); handler.mine(1)
+
+    assert vmc_handler.sample(shard_id) != zero_addr
+    assert vmc_handler.get_num_validators() == 1
+    print("!@# get_num_validators(): ", vmc_handler.get_num_validators())
+
+    genesis_colhdr_hash = b'\x00' * 32
+    header1 = get_testing_colhdr(vmc_handler, shard_id, genesis_colhdr_hash, 1)
+    header1_hash = sha3(header1)
+    vmc_handler.add_header(header1, primary_addr)
+    vmc_handler.chain_handler.mine(SHUFFLING_CYCLE_LENGTH)
+
+    header2 = get_testing_colhdr(vmc_handler, shard_id, header1_hash, 2)
+    header2_hash = sha3(header2)
+    vmc_handler.add_header(header2, primary_addr)
+    vmc_handler.chain_handler.mine(SHUFFLING_CYCLE_LENGTH)
+
+    assert vmc_handler.get_collation_header_score(shard_id, header1_hash) == 1
+    assert vmc_handler.get_collation_header_score(shard_id, header2_hash) == 2
+
+    vmc_handler.tx_to_shard(
+        keys[1].public_key.to_checksum_address(),
+        shard_id,
+        100000,
+        1,
+        b'',
+        1234567,
+        primary_addr,
+    )
+    vmc_handler.chain_handler.mine(1)
+    assert vmc_handler.get_receipt_value(0) == 1234567
+
+    do_withdraw(vmc_handler, validator_index)
+    vmc_handler.chain_handler.mine(1)
+    assert vmc_handler.sample(shard_id) == zero_addr
+
+if __name__ == '__main__':
```

Comment:
This should be removed as these tests should be runnable using `$ py.test tests/sharding/test_vmc_handler.py`

Reply:
Yep sorry it's my fault. I forgot to remove it.